src/Altinn.Platform/Altinn.Platform.Authentication/Authentication/Controllers/AuthenticationController.cs

```diff
@@ -340,8 +343,58 @@ private async Task<ActionResult> AuthenticateMaskinportenToken(string originalTo
+                {
+                    string enterpriseUserHeader = Request.Headers["X-Altinn-EnterpriseUser-Authentication"];
+                    byte[] decodedCredentials = Convert.FromBase64String(enterpriseUserHeader);
+                    string decodedString = Encoding.UTF8.GetString(decodedCredentials);
+                    string[] decodedStringArray = decodedString.Split(":");
+                    string usernameFromRequest = decodedStringArray[0];
+                    string password = decodedStringArray[1];
+
+                    EnterpriseUserCredentials credentials = new EnterpriseUserCredentials { UserName = usernameFromRequest, Password = password, OrganizationNumber = orgNumber };
+                    HttpResponseMessage response = await _enterpriseUserAuthenticationService.AuthenticateEnterpriseUser(credentials);
+                    string content = await response.Content.ReadAsStringAsync();
+
+                    switch (response.StatusCode)
+                    {
+                        case System.Net.HttpStatusCode.BadRequest:
+                            return StatusCode(400);
+                        case System.Net.HttpStatusCode.NotFound:
+                            ObjectResult result = StatusCode(404, "The user either does not exist or the password is incorrect.");
+                            return result;
+                        case System.Net.HttpStatusCode.InternalServerError:
+                            return StatusCode(502);
+                        case System.Net.HttpStatusCode.TooManyRequests:
+                            if (response.Headers.RetryAfter != null)
+                            {
+                                Response.Headers.Add("Retry-After", response.Headers.RetryAfter.ToString());
+                            }
+
+                            return StatusCode(429);
+                        case System.Net.HttpStatusCode.OK:
+                            authenticatemethod = "virksomhetsbruker";
+
+                            UserAuthenticationResult userAuthenticationResult = new UserAuthenticationResult();
+                            userAuthenticationResult = JsonSerializer.Deserialize<UserAuthenticationResult>(content);
+
+                            string userID = userAuthenticationResult.UserID.ToString();
+                            string username = userAuthenticationResult.Username;
+                            string partyId = userAuthenticationResult.PartyID.ToString();
+
+                            claims.Add(new Claim(AltinnCoreClaimTypes.UserId, userID, ClaimValueTypes.Integer32, issuer));
+                            claims.Add(new Claim(AltinnCoreClaimTypes.UserName, username, ClaimValueTypes.String, issuer));
+                            claims.Add(new Claim(AltinnCoreClaimTypes.PartyID, partyId, ClaimValueTypes.Integer32, issuer));
+                            break;
+                        default:
+                            return StatusCode(401);
```

Comment:
Replace with StatusCode(502). Should also include something like this to aid debugging/monitoring:

`_logger.LogWarning("Unexpected response from SBLBridge during enterprise user authentication. HttpStatusCode={statusCode} Content={content}", response.StatusCode, content);`